src/oc/interaction/api/comments.clj

```diff
@@ -26,8 +32,96 @@
     (catch clojure.lang.ExceptionInfo e
       [false, {:reason (.getMessage e)}]))) ; Not a valid new interaction
 
+(defn- valid-comment-update? [conn ctx comment-uuid comment-props]
+  (if-let [existing-comment (or (:existing-comment ctx) (interact-res/get-interaction conn comment-uuid))]
+    ;; Merge the existing comment with the new updates
+    (let [updated-comment (merge existing-comment (interact-res/clean comment-props))]
+      (if (lib-schema/valid? interact-res/Comment updated-comment)
+        {:existing-comment existing-comment :updated-comment updated-comment}
+        [false, {:updated-comment updated-comment}])) ; invalid update
+    
+    true)) ; no existing comment, so this will fail existence check later
+
+(defn author-match?
+  "Determine if the comment specified by the UUID was authored by the current user."
+  [conn ctx comment-uuid]
+  (if-let* [user-id (-> ctx :user :user-id)
+            existing-comment (or (:existing-comment ctx) (interact-res/get-interaction conn comment-uuid))
+            author-id (-> existing-comment :author :user-id)]
+    (if (= user-id author-id)
```

Comment:
Right. Not sure if we'll ever let someone else edit or delete it, but we won't for now unless/until a customer brings it up and wants it.